File: server/data/results/translations/lite/plain/go/run_length_encode.go
package main

import (
	"fmt"
	"strconv"
)

func encode(text string) string {
	if len(text) == 0 {
		return ""
	}
	result := ""
	current := rune(text[0])
	count := 1
	i := 1
	for i < len(text) {
		if rune(text[i]) == current {
			count = count + 1
		} else {
			result = result + string(current) + strconv.Itoa(count)
			current = rune(text[i])
			count = 1
		}
		i = i + 1
	}
	result = result + string(current) + strconv.Itoa(count)
	return result
}

func main() {
	samples := []string{"aaabbc", "abcd", "zzzzzzzz"}
	for _, sample := range samples {
		fmt.Println(encode(sample))
	}
}

File: server/data/results/translations/lite/plain/javascript/run_length_encode.js
function encode(text) {
    if (text.length === 0) {
        return "";
    }
    let result = "";
    let current = text[0];
    let count = 1;
    let i = 1;
    while (i < text.length) {
        if (text[i] === current) {
            count = count + 1;
        } else {
            result = result + current + String(count);
            current = text[i];
            count = 1;
        }
        i = i + 1;
    }
    result = result + current + String(count);
    return result;
}

function main() {
    const samples = ["aaabbc", "abcd", "zzzzzzzz"];
    for (const sample of samples) {
        console.log(encode(sample));
    }
}

main();